verifiers/internal/gcb/provenance.go

```diff
@@ -166,11 +151,23 @@ func (p *Provenance) VerifyTextProvenance() error {
 		return err
 	}
 
-	// Note: there is an additional field `metadata.buildInvocationId` which
-	// is not part of the specs but is present. This field is currently ignored during comparison.
-	unverifiedTextIntotoStatement := v01IntotoStatement{
-		StatementHeader: p.verifiedProvenance.Build.UnverifiedTextIntotoStatement.StatementHeader,
-		Predicate:       p.verifiedProvenance.Build.UnverifiedTextIntotoStatement.SlsaProvenance,
+	statement := p.verifiedIntotoStatement
+	predicate, err := (*statement).Predicate()
+	if err != nil {
+		return err
+	}
+
+	var unverifiedTextIntotoStatement interface{}
+	switch v := predicate.(type) {
+	case v01.ProvenancePredicate:
+		// Note: there is an additional field `metadata.buildInvocationId` which
```

Comment:
nit: uppercase NOTE labels
```suggestion
		// NOTE: there is an additional field `metadata.buildInvocationId` which
```